### PR TITLE
bradl3yC - 5866 - Screen reader re-reading previous modal

### PR DESCRIPTION
**Pull request by bradl3yC**
## Description
- [x] - close modal before opening the next 

## Screenshots
Screen Shot 2020-02-25 at 4 34 18 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx [hunk 1]
```diff
@@ -182,14 +187,13 @@ class AddressValidationModal extends React.Component {
...
       addressValidationError,
-      closeModal,
       resetAddressValidation,
       confirmedSuggestions,
     } = this.props;
 
     const resetDataAndCloseModal = () => {
       resetAddressValidation();
-      closeModal();
+      this.props.closeModal();
```
Comment (bradl3yC) [inline src/platform/user/profile/vet360/containers/AddressValidationModal.jsx]: for some reason, none of the dispatches to close the modal were showing up in redux dev tools - by switching to this pattern, they now do.

**Review by LinnJS — APPROVED** [review]

**Review by LinnJS — APPROVED** [review]

**Review by bkjohnson — COMMENTED** [review]
The code looks fine, but I don't really understand what it is solving.  Was there a user workflow that would cause two modals to be opened at once?

**Comment by bradl3yC** [comment]
@bkjohnson great question! So basically what was happening is the currently opened modal was receiving a prop update BEFORE it closed.  So the screen reader would re-read its content before moving on to the newly opened modal.  The old action would overwrite the name of the currently open modal with the new modal thus closing one and opening the other.  What this does is just call close modal before opening the new modal - It should prevent the screen reader from re-reading the content from the currently opened modal.

**Review by bkjohnson — APPROVED** [review]
Thank you for the explanation!  My only thought is that it might be nice to have an inline comment explaining why a button click is closing a modal before opening it, or maybe a unit test.  As someone who is unfamiliar with this specific code, it doesn't seem intuitive to me, so I can imagine that if a new engineer gets onboarded to the team a few months from now and sees this they may face similar confusion.  Additionally, without any "obvious" reason why it is there, it may get accidentally removed in a future modification, or a new engineer may copy the pattern to use in new code and not include it.

But I'm saying all of this with very little context, so take it with a grain of salt.

**Pull request merged**